pkg/server/commands/reverseexpand/reverse_expand_weighted.go

```diff
@@ -228,10 +237,38 @@ func (c *ReverseExpandQuery) loopOverEdges(
 				_, newStack := stack.Pop(newReq.relationStack)
 				newStack = stack.Push(newStack, typeRelEntry{typeRel: toNode.GetUniqueLabel()})
 				newReq.relationStack = newStack
+
+				pool.Go(func(ctx context.Context) error {
+					return c.dispatch(ctx, newReq, resultChan, needsCheck, resolutionMetadata)
+				})
+			} else {
```

Comment:
Instead of using an else here, is there an opportunity to break within the if above so we can exist the switch and reduce complexity?